Initial script for running tests

philippecamacho opened the pull request:
### This PR:

Adds a script to run the tests. For now many failing tests are skipped. We will gradually fix the code so that more tests can pass.
Also there is a README_ESPRESSO.md file that documents how to setup the development environment. It will be extended as well, in particular with the instructions described in https://www.notion.so/espressosys/Local-Test-Setup-1a62431b68e9806e8f62c77a98d4a5e2.


### How to test this PR:  

Follow the instructions in `README_ESPRESSO.md`. All the tests should pass.

philippecamacho committed:
Initial script to run the tests

dailinsubjam commented:
The command works good for me, but I'll leave the approval to people more familiar with shell.

Ayiga committed:
Update the run_all_tests.sh file

The run_all_tests.sh file was initially setup from a prior integration
and potentially reflects a project state that no longer exists.  For
example `proxyd` and `indexer` no longer seem to be folders that
exist in the project.  Additionally, there are a few complex commands
to run all of the `test` make file commands. However not all of these
tests are run in the same way. Additionally, these tests are not
necessarily run in isolation from each other.  There are
a few tests that `test-unit` run that require the existence of a
`forge-artifacts` file which can be populated by another test.

This commit attempts to specify all of the tests in an unambiguous
sequential  manner with the correct `Makefile` command, while
keeping the early exit for easier test inspection.

philippecamacho commented:
Thanks @Ayiga for improving the PR!
